src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -910,6 +910,9 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             //      becomes a no-op.
             this->Focus(FocusState::Programmatic);
 
+            // Now that the renderer is set up, update the appearance for initialization
+            UpdateAppearance(_settings);

```

Comment:
I am *still* deeply uncomfortable with us going from the main thread (initializing), under lock, and pushing another block onto the main thread queue that takes the lock and updates the appearance.

This is all stuff that could be done directly on the main thread without the intermediate step-off-step-back-onto-the-main-thread business.

Is the only problem that UpdateAppearance->UpdateAppearanceOnMainThread _takes the lock_ that is held in this function? If so, make it `_UpdateAppearanceOnMainThreadUnderLock` and make sure that all callers take the correct lock before calling it.

Reply:
I'm uncomfortable with it because we're introducing a point in the middle, during initialization, where somebody else could get code running on the main thread and manipulate the state of the terminal. Imagine us losing focus between Initialize and when UpdateAppearance finally runs -- we'll flicker to unfocused appearance and then THIS UpdateAppearance will finally trigger and flip it back to focused¹


¹ assuming that the focus change is scheduled in between [Init] and [UpdateAppearance(settings)]
² unless the focus handler itself dispatches BACK to the main thread (even though it started there), then the queue becomes... `Init ; Handle Focus ; UpdateAppearance(settings) ; UpdateAppearance(unfocusedSettings)` (which _works_, but... why)

Reply:
Thank you for the explanation! Changed to `_UpdateAppearanceOnUIThreadUnderLock` and `_UpdateSettingsOnUIThreadUnderLock`, and the callers take the lock before calling it